### PR TITLE
Add Codyze v3 while keeping Codyze v2

### PR DESCRIPTION
**Pull request by maximilian-galanis**

### CODE REVIEW

**Comment by oxisto** [comment]
this somehow adds 13.000 new LoC? Seems like something is not right here :D

**Comment by maximilian-galanis** [comment]
> 

The current codyze_v3/codyze-specification-languages/mark contains the majority of the codyze_v2 code.

File: codyze_v2/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+```shell
+$ ./gradlew installDist
+```
+
+This will provide you with an executable Codyze installation under `build/install/codyze`.
+To start Codyze, change to the directory and run Codyze.
+
+Codyze has three execution modes:
+* commando line interface mode (`-c`, default)
```
Comment (KuechA) [inline codyze_v2/README.md:36]:
```suggestion
* command line interface mode (`-c`, default)
```

**Comment by seelchen** [comment]
> Should we filter out duplicate passes?

In V2 we didn't filter them out. Since in the CPG the previous passes directly affect the result of the following passes, there might be cases where duplicate passes are needed, but I'm not completely sure...

**Comment by maximilian-galanis** [comment]
> I'm wondering what's the difference between the package `de.fraunhofer.aisec.codyze.legacy.*` and moving all the code to a subdirectory `codyze_v2`. I thought the purpose of adding the legacy package was to keep all code of Codyze2 there and add new code somewhere else. This should be the same as the goal of codyze_v2 or not?
> 
> Will we keep the usage (i.e., the parameters of the CLI and config) the same for v2 and v3?

The new project layout is a [gradle composite build](https://docs.gradle.org/current/userguide/composite_builds.html), which allows us to develop both v2 and v3 independently.

Regarding the usage, v3 still allows for both CLI and a config file and exhibits the same behavior (e.g., CLI options overwrite values in the config file). It also allows for pretty much the same options for now. For the sake of clarity, however, v3 renames some options and changes how exactly options must be provided or are processed. This also extends to the layout of the config file. While .yaml config files will still be supported, v3 expects a slightly different layout to the config file.

**Review by oxisto — APPROVED** [review]

**Comment by oxisto** [comment]
Finally passes all CI checks. One minor thing. I would probably personally prefer `codyze-v2` instead of `codyze_v2` since kebab case is more commonly used in the JVM/Kotlin world than snake case.

**Pull request merged**